packages/terminal-manager/src/browser/terminal-manager-tree-model.ts

```diff
@@ -0,0 +1,349 @@
+
+    protected onTerminalGroupAddedEmitter = new Emitter<{
+        groupId: TerminalManagerTreeTypes.GroupId,
+        pageId: TerminalManagerTreeTypes.PageId,
+        terminalKey: TerminalManagerTreeTypes.TerminalKey,
+    }>();
+
+    readonly onTerminalGroupAdded = this.onTerminalGroupAddedEmitter.event;
+    protected onTerminalGroupDeletedEmitter = new Emitter<TerminalManagerTreeTypes.GroupId>();
+    readonly onTerminalGroupDeleted = this.onTerminalGroupDeletedEmitter.event;
+
+    protected onTerminalAddedToGroupEmitter = new Emitter<{
+        terminalId: TerminalManagerTreeTypes.TerminalKey,
+        groupId: TerminalManagerTreeTypes.GroupId,
+    }>();
+
+    readonly onTerminalAddedToGroup = this.onTerminalAddedToGroupEmitter.event;
+    protected onTerminalDeletedFromGroupEmitter = new Emitter<{
+        terminalId: TerminalManagerTreeTypes.TerminalKey,
+        groupId: TerminalManagerTreeTypes.GroupId,
+    }>();
+
+    readonly onTerminalDeletedFromGroup = this.onTerminalDeletedFromGroupEmitter.event;
+
+    @postConstruct()
+    protected override init(): void {
+        super.init();
+        this.toDispose.push(this.selectionService.onSelectionChanged(selectionEvent => {
+            const selectedNode = selectionEvent.find(node => node.selected);
+            if (selectedNode) {
+                this.handleSelectionChanged(selectedNode);
+            }
+        }));
+        this.root = { id: 'root', parent: undefined, children: [], visible: false } as CompositeTreeNode;
+    }
+
+    protected getContext = () => this;
+
+    addTerminalPage(
+        terminalKey: TerminalManagerTreeTypes.TerminalKey,
+        groupId: TerminalManagerTreeTypes.GroupId,
+        pageId: TerminalManagerTreeTypes.PageId,
+    ): void {
+        const pageNode = this.createPageNode(pageId);
+        const groupNode = this.createGroupNode(groupId, pageId);
+        const terminalNode = this.createTerminalNode(terminalKey, groupId);
+        if (this.root && CompositeTreeNode.is(this.root)) {
+            this.activePageNode = pageNode;
+            CompositeTreeNode.addChild(groupNode, terminalNode);
+            CompositeTreeNode.addChild(pageNode, groupNode);
+            this.root = CompositeTreeNode.addChild(this.root, pageNode);
+            this.onPageAddedEmitter.fire({ pageId: pageNode.id, terminalKey });
+            setTimeout(() => {
+                this.selectionService.addSelection(terminalNode);
+            });
+        }
+    }
+
+    getName(): string {
+        const pageLabel = this.activePageNode?.label;
+        const groupLabel = this.activeGroupNode?.label;
+        const terminalLabel = this.activeTerminalNode?.label;
+        let name = '';
+        if (pageLabel) {
+            name += pageLabel;
+        }
+        if (groupLabel) {
+            name += ` > ${groupLabel}`;
+        }
+        if (terminalLabel) {
+            name += ` > ${terminalLabel}`;
+        }
+        return name;
+    }
```

Comment:
I don't see any uses of this method(?), and if it were used, it should probably go in a label provider.